agent/api/task.go

```diff
@@ -186,13 +186,22 @@ func TaskFromACS(acsTask *ecsacs.Task, envelope *ecsacs.PayloadMessage) (*Task,
 // PostUnmarshalTask is run after a task has been unmarshalled, but before it has been
 // run. It is possible it will be subsequently called after that and should be
 // able to handle such an occurrence appropriately (e.g. behave idempotently).
-func (task *Task) PostUnmarshalTask(cfg *config.Config, credentialsManager credentials.Manager) {
+func (task *Task) PostUnmarshalTask(cfg *config.Config,
+	credentialsManager credentials.Manager, resourceFields taskresource.ResourceFields) error {
 	// TODO, add rudimentary plugin support and call any plugins that want to
 	// hook into this
 	task.adjustForPlatform(cfg)
+	if task.MemoryCPULimitsEnabled {
```

Comment:
can this happen within 'adjustForPlatform' ?

Reply:
Made that a separate method to return a particular error type `ResourceInitializeError` when err from `initializeCgroupResource` is not nil.